config/v1/types_feature.go

```diff
@@ -112,6 +112,16 @@ type FeatureGateStatus struct {
 	// +listType=map
 	// +listMapKey=version
 	FeatureGates []FeatureGateDetails `json:"featureGates"`
+
```

Comment:
We wanted to include a `generation` field in here that would be monotonically increasing every time the `featureGates`/`renderedMinimumComponentVersions` list changes

Reply:
I don't currently see this as necessary actually . How would the handling be different than a user manually changing the featureset in a custom no upgrade case?

Reply:
Yeah, it would be about the same. The intention of the generation was that we thought we'd be able to use it to show the MCP had rolled out the latest observed generation of the feature gates. What signal will we use to check the MCP has rolled out the latest version of these changes?

Reply:
my current approach is using the MCP [updated condition](https://github.com/openshift/machine-config-operator/pull/4929/files#diff-935aab0fbf5801d5f98c5670c9f242c0c70a5bb264976d15f9d83db454534e38R773)  similar to how the [MCS serves the config](https://github.com/openshift/machine-config-operator/blob/main/pkg/server/cluster_server.go#L122), because the MCS serving in the config will be how a lower versioned node gets its config, so after the new config is rendered it should be safe for a lower version node to be created

Reply:
Is there a race condition that you need to be careful of here? The feature gates get updated, the MCP condition says it's up to date, you continue, the MCP controller then realises the FG change and marks the updated condition false?

How do we ensure that the MCP controller has seen the change before we observe the updated condition?

Reply:
yeah I have to get some advice from MCO team on that, but there is an ObservedGeneration field in the mcp status I intend on checking